codice/ddf
# DDF-04299: Refactored 'result-item.view' to allow for lazy thumbnail load

#### What does this PR do?
Makes thumbnails load on-demand (or when grid/gallery view is selected). Also adds image source fallback...when image URL is not available default icon is displayed (pictures attached).
#### Who is reviewing it? 
@brianfelix 
@Corey-Collins 
@jrnorth 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@millerw8

#### How should this be tested?
1. Open `Developer Tools` (Chrome)
2. Search for results
3. If `List` view is selected (and no table, inspector, etc. is open) observe no thumbnails are fetched
4. Switch to `Gallery` view
5. Observe thumbnails being fetched

#### Any background context you want to provide?
#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #4299 

#### Screenshots
![thumbnail-lazy-load](https://user-images.githubusercontent.com/118054/52128901-2d1b4f80-2604-11e9-9b0f-9c36a1013f1e.gif)

Before image fallback:
<img width="1792" alt="screen shot 2019-02-01 at 1 56 50 pm" src="https://user-images.githubusercontent.com/118054/52152160-62935d80-2643-11e9-9145-9a0eb1c5ac08.png">

After image fallback:
<img width="1792" alt="screen shot 2019-02-01 at 1 55 20 pm" src="https://user-images.githubusercontent.com/118054/52152170-7048e300-2643-11e9-8648-d9620b974c9a.png">

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
